### DISCUSSION TITLE
How to setup Remark42 Instance on IIS with Binary?

### DISCUSSION THREAD

**Discussion by Line4GitHub** [umputun/remark42#1195] [General]
How to setup Remark42 Instance on Internet Information Services (IIS) for Windows with Binary?
Are there step by step tutorial?

**Comment by paskal** [comment]
As of now, the only way to get such a tutorial is to write it yourself, as you are the first person ever mentioning IIS in the issues/discussions of remark42.

**Reply by Line4GitHub** [reply]
Thank you very much for your reply.

**Reply by paskal** [reply]
I have a backlog item for myself to write a step-by-step tutorial for the installation of remark42. As soon as I get to it, I'll look into writing one for IIS as well, as it should not be that hard. For now, I assume the only way to move forward is to try to match the installation path for Nginx to IIS, which requires quite a bit of knowledge in both.

**Reply by Line4GitHub** [reply]
Thank you very much for your clarification.
And very sorry for my poor English!

I follow the instructions in  [this tutorial](https://www.oqtane.org/Resources/Blog/PostId/542/installing-oqtane-on-iis) to set up the website smoothly, Remark42 can also be displayed smoothly, if
<pre>// E:\inetpub\Remark42\index.html
host: 'https://demo.remark42.com',
url: 'https://remark42.com/demo/',
</pre>
<hr>
<p>but</P>
<pre>
host: 'http://127.0.0.1:8080/',
url: 'http://127.0.0.1:8080/web/',
</pre>
<hr>
<pre>
E:\cd E:\inetpub\Remark42
E:\inetpub\Remark42>remark42.windows-amd64 server /secret:12345 /url:http://127.0.0.1:8080
remark42 v1.8.1-82f8f6c9-20210504T18:25:06
2021/12/19 18:57:39.324 [INFO]  start server on port 8080
2021/12/19 18:57:39.325 [INFO]  root url=http://127.0.0.1:8080
2021/12/19 18:57:39.325 [INFO]  make data store, type=bolt
2021/12/19 18:57:39.327 [INFO]  bolt store for sites [{FileName:./var/remark.db SiteID:remark}], options {Timeout:30s NoGrowSync:false NoFreelistSync:false FreelistType: ReadOnly:false MmapFlags:0 InitialMmapSize:0 PageSize:0 NoSync:false OpenFile:<nil>}
2021/12/19 18:57:39.382 [INFO]  make admin store, type=shared
2021/12/19 18:57:39.382 [INFO]  make cache, type=mem
2021/12/19 18:57:39.384 [INFO]  make avatar store, type=fs
2021/12/19 18:57:39.385 [WARN]  no auth providers defined
2021/12/19 18:57:39.386 [INFO]  activate http rest server on port 8080
2021/12/19 18:57:39.386 [INFO]  activate auto-backup for remark under ./var/backup, duration 24h0m0s
2021/12/19 18:57:39.386 [INFO]  start pictures cleanup, staging ttl=12m30s
2021/12/19 18:57:39.389 [INFO]  templates will be read from statik
2021/12/19 18:57:39.490 [INFO]  run file server for ./web, embedded
</pre>
<pre>
// in browser nav (address) bar input 
http://127.0.0.1:8080/
// result in browser
404 page not found
</pre>
<pre>
// E:\inetpub\Remark42\index.html & E:\inetpub\Remark42\web\index.html
&lt;div id="remark42" class="dark:prose-dark" style="width: 80%; margin-left: 10%;">
&lt;/div>

&lt;noscript>You use your browser with disabled JavaScript. Please enable JavaScript for comments.&lt;/noscript>

&lt;script>window.remark_config = {
		host: 'http://127.0.0.1:8080/',
		site_id: 'remark',
		//url: 'http://127.0.0.1:8080/web',
        url: window.location.href,
		components: ['embed'],
		max_shown_comments: 20,
		locale: 'zh',
		theme: document.documentElement.classList.contains('dark') ? 'dark' : 'light'
	}&lt;/script>

&lt;script>!function(e,n){for(var o=0;o&lt;e.length;o++){var r=n.createElement("script"),c=".js",d=n.head||n.body;"noModule"in r?(r.type="module",c=".mjs"):r.async=!0,r.defer=!0,r.src=remark_config.host+"/web/"+e[o]+c,d.appendChild(r)}}(remark_config.components||["embed"],document);&lt;/script>
</pre>
